src/transformers/generation_utils.py

```diff
@@ -484,13 +485,11 @@ def _expand_inputs_for_generation(
         return input_ids, model_kwargs
 
     @staticmethod
-    def _init_sequence_length_for_generation(
-        input_ids: torch.LongTensor, max_length: int
-    ) -> Tuple[torch.Tensor, torch.Tensor, int]:
+    def _init_sequence_length_for_generation(input_ids: torch.LongTensor) -> Tuple[torch.Tensor, torch.Tensor, int]:
```

Comment:
I think [this PR](https://github.com/huggingface/transformers/pull/11331) will resolve the issue with `max_length` so that we can maybe merge it first - sorry for the merge conflict :-/